### PR TITLE
[v1.16] Author proxy backports

### PR DESCRIPTION
**Pull request by jrajahalme**
* [ ] #36142 (@jrajahalme)
 * [ ] #36389 (@jrajahalme)
 * [ ] #36399 (@jrajahalme)
 * [ ] #36435 (@jrajahalme)
 * [ ] #36330 (@jrajahalme) ⚠️ had to add a commit to update github.com/cilium/proxy Envoy API, which also pulled in various other vendor updates
 
Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 36142 36389 36399 36435 36330
```

### CODE REVIEW

**Comment by jrajahalme** [comment]
/test-backport-1.16

**Comment by jrajahalme** [comment]
Fixed vendoring

**Comment by jrajahalme** [comment]
Added new vendored files to the 2nd to last commit.

**Comment by jrajahalme** [comment]
/test-backport-1.16

**Comment by jrajahalme** [comment]
/test-backport-1.16

**Comment by jrajahalme** [comment]
/test-backport-1.16

File: go.mod [hunk 1]
```diff
@@ -1,7 +1,9 @@
 module github.com/cilium/cilium
 
 // renovate: datasource=golang-version depName=go
-go 1.22.0
+go 1.22.7
+
+toolchain go1.22.10
```
Comment (aanm) [inline go.mod]: These lines look suspicious, it should probably be reverted.

**Review by aanm — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (tklauser) [inline go.mod]: Please drop this hunk. We want to keep the Go version at the .0 patch release so we do not force developers to upgrade Go patch versions. The toolchain version can be dropped as well.

**Review by tklauser — CHANGES_REQUESTED** [review]

**Comment by jrajahalme** [comment]
/test-backport-1.16

**Review by aanm — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (jrajahalme) [reply go.mod]: reverted an dependency update on cilium/proxy v1.30, now this is gone.

Comment (jrajahalme) [reply go.mod]: Fixed :-)

**Review by tklauser — APPROVED** [review]

**Pull request merged**